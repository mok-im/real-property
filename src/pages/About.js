import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { Carousel, Container, Row, Col, ListGroup } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { fetchOneHouse } from '../redux/module/reducer';
import styled from 'styled-components';

const ImgHolder = styled.div`
    height: 40rem;
    display: flex;
    align-items: center;
    background-color: #282828;
`;

const About = () => {
    const houses = useSelector((state) => state.houses);
    const { id } = useParams();
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(fetchOneHouse(+id));
    }, []);


    const renderHouse = houses.map((house) =>
        <Row xs={1} sm={1} md={2}>
            <Col >
                <Carousel fade interval={null}>
                    {
                        house.img_collection.map((img) =>
                            <Carousel.Item>
                                <ImgHolder >
                                    <img
                                        fluid
                                        className="d-block w-100"
                                        src={img.img}
                                        alt={img.id}
                                    />
                                </ImgHolder>
                            </Carousel.Item>
                        )
                    }
                </Carousel>
            </Col>
            <Col  >
                <ListGroup >
                    <ListGroup.Item>Sales name: {house.sales_name}  </ListGroup.Item>
                    <ListGroup.Item>Name House: {house.name_house}</ListGroup.Item>
                    <ListGroup.Item>Price: {house.price}</ListGroup.Item>
                    <ListGroup.Item>Address: {house.address}</ListGroup.Item>
                    <ListGroup.Item>Description: {house.description}</ListGroup.Item>
                </ListGroup>
            </Col>
        </Row>
    )

    return (
        <Container className="bg-light pt-2 pb-2 rounded">
            {renderHouse}
        </Container>
    );
};

export default About;