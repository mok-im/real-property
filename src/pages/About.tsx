import React, { useEffect } from 'react';
import { NavLink, useParams } from 'react-router-dom';
import { Carousel, Container, Row, Col, ListGroup } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { fetchOneHouse } from '../redux/ducks/actions';
import { useTypedSelector } from '../hooks/useTypedSelector';
import { IParams, IImg } from '../types';
import { ImgHolder } from '../styled/ImgHolder';

const About: React.FC = () => {
    const houses = useTypedSelector((state) => state.house);
    const load = useTypedSelector((state) => state.loading);
    const { id } = useParams<IParams>();
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(fetchOneHouse(+id));
    }, [dispatch, id]);

    return (
        <Container>
            <NavLink className="btn btn-primary mb-4" to={'/'}>
                &larr; Back
            </NavLink>
            {load ? (
                <h1>Loading...</h1>
            ) : (
                houses.map((house, i) => (
                    <Row xs={1} sm={1} md={2} key={i}>
                        <Col>
                            <Carousel fade interval={null}>
                                {house.img_collection.map(
                                    (img: IImg, index: string) => (
                                        <Carousel.Item key={index}>
                                            <ImgHolder>
                                                <img
                                                    className="d-block w-100"
                                                    src={img.img}
                                                    alt="house"
                                                />
                                            </ImgHolder>
                                        </Carousel.Item>
                                    )
                                )}
                            </Carousel>
                        </Col>
                        <Col>
                            <ListGroup>
                                <ListGroup.Item>
                                    Sales name: {house.sales_name}
                                </ListGroup.Item>
                                <ListGroup.Item>
                                    Name House: {house.name_house}
                                </ListGroup.Item>
                                <ListGroup.Item>
                                    Price: {house.price}
                                </ListGroup.Item>
                                <ListGroup.Item>
                                    Address: {house.address}
                                </ListGroup.Item>
                                <ListGroup.Item>
                                    Description: {house.description}
                                </ListGroup.Item>
                            </ListGroup>
                        </Col>
                    </Row>
                ))
            )}
        </Container>
    );
};

export default About;
