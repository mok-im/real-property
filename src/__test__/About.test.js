import React from 'react';
import { shallow } from 'enzyme';
import About from '../pages/About';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const state = {
    house: [
        {
            id: 1,
            name_house: "House 1",
            price: "1000$",
            address: "New York City1",
            img: "https://ap.rdcpix.com/ce4305635b3d44aef2e6abe0ce8f1f93l-m507177317od-w1024_h768_x2.webp",
            description: "Lorem1 ",
            sales_name: "John1",
            img_collection: [
                {
                    id: 1,
                    img: "https://ap.rdcpix.com/ce4305635b3d44aef2e6abe0ce8f1f93l-m1955241116od-w1024_h768.webp"
                }
            ]
        }
    ],
    loading: true,
}

let store;
const setUp = (state) => shallow(
    <Provider store={store}>
        <About {...state} />
    </Provider>);

describe('Testing page About', () => {
    it('should render page About', () => {
        const mockStore = configureStore();
        store = mockStore(state);
        const house = setUp(store);
        expect(toJson(house)).toMatchSnapshot();
    });

});

