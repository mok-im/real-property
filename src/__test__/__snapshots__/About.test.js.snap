// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing page About should render page About 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <About
    clearActions={[Function]}
    dispatch={[Function]}
    getActions={[Function]}
    getState={[Function]}
    replaceReducer={[Function]}
    subscribe={[Function]}
  />
</ContextProvider>
`;
